javalib/src/main/scala/java/util/SplittableRandom.scala

```diff
@@ -0,0 +1,122 @@
+package java.util
+
+/*
+ * This is a clean room implementation derived from the original paper
+ * and Java implementation mentioned there:
+ *
+ * Fast Splittable Pseudorandom Number Generators
+ * by Guy L. Steele Jr., Doug Lea, Christine H. Flood
+ * http://gee.cs.oswego.edu/dl/papers/oopsla14.pdf
+ *
+ */
+private object SplittableRandom {
+
+  private final val DoubleULP = 1.0 / (1L << 53)
+  private final val GoldenGamma = 0x9e3779b97f4a7c15L // odd
+
+  private var defaultGen: Long = new Random().nextLong()
+
+  private def nextDefaultGen(): Long = {
+    val s = defaultGen
+    defaultGen = s + (2 * GoldenGamma)
+    s
+  }
+
+  case class GenLong(value: Long)
+
+  private final def mix64ForGamma(z: Long): Long = {
```

Comment:
This method is missing its comment saying what it implements. IIRC that's the original MurmurHash 3 finalizer.